maths/extended_gcd.nim

```diff
@@ -0,0 +1,65 @@
+## Extended Euclidean algorithm
+##
+## [Extended Euclidean algorithm](https://en.wikipedia.org/wiki/Extended_Euclidean_algorithm)
+
+import std/math
+
+
+func euclidIteration(a, b: int): (int, int, int) =
+    assert a >= 0 and b >= 0
+    if b == 0:
+        return (a, (if a == 0: 0 else: 1), 0)
+    let (gcd, x, y) = euclidIteration(b, a mod b)
+    return (gcd, y, x - (a div b) * y)
```

Comment:
```suggestion
    if b == 0:
      (a, (if a == 0: 0 else: 1), 0)
    else:
      let (gcd, x, y) = euclidIteration(b, a mod b)
      (gcd, y, x - (a div b) * y)
```
You can also leave the first early return and do without the else branch but I'd say the suggested version is preferable as it's immediately obvious that the second block is only valid when the condition isn't triggered. 

See [Result return]( https://github.com/TheAlgorithms/Nim/blob/main/CONTRIBUTING.md#result-return) part of the guidelines.